index.js

```diff
@@ -16,12 +16,12 @@ var regex = new RegExp(pattern, 'm')
 module.exports = extractor
 module.exports.test = test
 
-function extractor (string) {
+function extractor (string, options = {allowUnsafe: false}) {
```

Comment:
Since this is the function that is exposed publicly it would be good to add some extra hygiene around argument checking/parsing to ensure there aren't errors from the internals of the library when users get a little loose with the input. Using part of the unix unix philosophy for inspiration: "write your software to cope with incorrect inputs and its own execution errors as gracefully as possible" 

To ensure we have an object to work with and a correct default the in case someone calls `fm('...', null)`

    options = options instanceof Object ? options : { allowUnsafe: false }

Note that the es6 syntax wont enforce this so it still has to be done manually.

Then to coerce `options.allowUnsafe` to a boolean:

    options.allowunsafe = Boolean(options.allowunsafe)

After that I think the input is sufficiently scrubbed :)

Reply:
Hang on, what are you suggesting? To change it to:
```javascript
function extractor (string, options) {
  string = string || ''
  options = options instanceof Object ? options : { allowUnsafe: false }
  options.allowunsafe = Boolean(options.allowUnsafe)
  ...
```
Wouldn't it be better with object spread notation:
```javascript
function extractor (string, options) {
  string = string || ''
  var defaultOptions = { allowUnsafe: false }
  options = options instanceof Object ? {...defaultOptions, ...options} : defaultOptions
  options.allowunsafe = Boolean(options.allowUnsafe)
  ...
```

That way, if we ever have non-trivial defaults (that work with `Boolean()`) you get the set things merged with the defaults. Like Python's keyword arguments work.